percona/controller/pgcluster/controller.go

```diff
@@ -524,15 +541,64 @@ func (r *PGClusterReconciler) handleMonitorUserPassChange(ctx context.Context, c
 	return nil
 }
 
-func (r *PGClusterReconciler) reconcileCustomExtensions(cr *v2.PerconaPGCluster) {
+func (r *PGClusterReconciler) reconcileCustomExtensions(ctx context.Context, cr *v2.PerconaPGCluster, postgresCluster *v1beta1.PostgresCluster) error {
+	log := logging.FromContext(ctx).WithValues("cluster", cr.Name, "namespace", cr.Namespace)
+
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(postgresCluster), postgresCluster); err != nil {
```

Comment:
You don't have to get `postgresCluster` again here, it is already retrieved from the API (at the beginning of the reconcile method) and you passed it to this method.